src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -1549,3 +1551,63 @@ def set_dataio(cls, **kwargs):
         """
         cargs, ckwargs = fmt_docval_args(H5DataIO.__init__, kwargs)
         return H5DataIO(*cargs, **ckwargs)
+
+    @docval(
```

Comment:
I think we should define the sidecar functionality in a separate class because: 


1. this is really a post-read operation that happens after the HDF5 I/O is complete
2. this is ultimately functionality that is not specific to HDF5 per se but will be useful for other I/O backends as well
3. it helps keep the already very complicated HDF5IO class a bit cleaner
4. makes it easier to maintain and enhance the sidecare functionality

Reply:
That makes a lot of sense

Reply:
Fixed